packages/hoppscotch-app/helpers/import-export/export/openapi.ts

```diff
@@ -0,0 +1,480 @@
+                    O.alt(() =>
+                      pipe(
+                        hoppRequestBody,
+                        generateOpenAPIBodyForPlainText,
+                        O.some
+                      )
+                    )
+                  )
+                )
+              )
+            ),
+            O.alt(() =>
+              pipe(
+                hoppRequestBody,
+                O.fromPredicate(isFormDataBody),
+                O.map(generateOpenAPIBodyForFormData)
+              )
+            ),
+            E.fromOption(() => "INVALID_BODY" as const)
+          )
+        )
+      )
+    )
+  )
+
+const OpenAPIAuthNames = {
+  basic: "basicAuth",
+  "api-key": "ApiKeyAuth",
+  bearer: "bearerAuth",
+} as const
+
+export const generateOpenApiAuth = (
+  hoppAuth: HoppRESTAuth
+): E.Either<"INVALID_AUTH", OpenAPIV3.SecurityRequirementObject> =>
+  pipe(
+    hoppAuth.authType,
+    O.fromPredicate((authType) => authType === "none"),
+    O.map(() => ({})),
+    O.alt(() =>
+      pipe(
+        hoppAuth.authType,
+        O.fromPredicate(
+          (authType): authType is "basic" | "api-key" | "bearer" =>
+            ["basic", "api-key", "bearer"].includes(authType)
+        ),
+        O.map((authType) =>
+          pipe(OpenAPIAuthNames, (auths) => ({
+            [auths[authType]]: [],
+          }))
+        )
+      )
+    ),
+    E.fromOption(() => "INVALID_AUTH" as const)
+  )
+
+const isValidMethod = (method: string): method is OpenAPIV3.HttpMethods =>
+  [
+    "get",
+    "post",
+    "patch",
+    "delete",
+    "options",
+    "head",
+    "patch",
+    "trace",
+  ].includes(method)
+
+const generateOpenApiDocument = (
+  paths: OpenAPIV3.PathsObject
+): OpenAPIV3.Document => ({
+  openapi: "3.0.0",
+  info: {
+    title: "Hoppscotch Openapi Export",
+    version: "1.0.0",
+  },
+  paths,
+  components: {
+    securitySchemes: {
+      basicAuth: {
+        type: "http",
+        scheme: "basic",
+      },
+      bearerAuth: {
+        type: "http",
+        scheme: "bearer",
+      },
+      ApiKeyAuth: {
+        type: "apiKey",
+        name: "api-key-header-name",
+        in: "header",
+      },
+    },
+  },
+})
+
+type PathGenerationError =
+  | "INVALID_METHOD"
+  | "INVALID_URL"
+  | "INVALID_CONTENT_TYPE"
+  | "INVALID_AUTH"
+  | RequestBodyGenerationError
+
+const generateOpenApiPathFromRequest = (
+  request: HoppRESTRequest
+): E.Either<
+  PathGenerationError,
+  OpenAPIV3.PathItemObject & { pathname: string }
+> =>
+  pipe(
+    E.Do,
+    E.bind("method", () =>
+      pipe(
+        request.method.toLowerCase(),
+        E.fromPredicate(isValidMethod, () => "INVALID_METHOD" as const)
+      )
+    ),
+    E.bindW("origin", () =>
+      pipe(
+        E.tryCatch(
+          () => new URL(request.endpoint),
+          () => "INVALID_URL" as const
+        )
+      )
+    ),
+
+    E.bindW("openApiBody", () =>
+      pipe(request.body, generateOpenApiRequestBody)
```

Comment:
You can drop pipes for small pipelines

```suggestion
    generateOpenApiRequestBody(request.body)
```